src/cluster.c

```diff
@@ -220,27 +234,39 @@ int clusterLoadConfig(char *filename) {
             n = createClusterNode(argv[0],0);
             clusterAddNode(n);
         }
-        /* Format for the node address information: 
-         * ip:port[@cport][,hostname] */
+        /* Format for the node address information:
+         * ip:port[@cport][,hostname],shard_id */
+
+        ep_argv = sdssplitlen(argv[1], sdslen(argv[1]), ",", 1, &ep_argc);
+        if (ep_argv == NULL) {
+            sdsfreesplitres(argv,argc);
+            goto fmterr;
+        }
 
         /* Hostname is an optional argument that defines the endpoint
          * that can be reported to clients instead of IP. */
-        char *hostname = strchr(argv[1], ',');
-        if (hostname) {
-            *hostname = '\0';
-            hostname++;
-            n->hostname = sdscpy(n->hostname, hostname);
+        if (ep_argc > 1 && sdslen(ep_argv[1]) > 0) {
+            n->hostname = sdscpy(n->hostname, ep_argv[1]);
         } else if (sdslen(n->hostname) != 0) {
             sdsclear(n->hostname);
         }
 
+        /* shard_id is the name of the shard this node belongs to.
+         * It is mandatory on 7.0 and above but can be absent right
+         * after the upgrade to 7.0+ */
+        if (ep_argc > 2) {
+            memcpy(n->shard_id, ep_argv[2], min(strlen(ep_argv[2]), CLUSTER_NAMELEN));
```

Comment:
I would throw an error if the value is not CLUSTER_NAMELEN and not load something random.